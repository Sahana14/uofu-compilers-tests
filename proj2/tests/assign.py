a = b
a, b = c
a, = c
a, b, = c
a = yield
a = yield from baz
a, b, c = yield
a, b, c = yield from baz
a = foo(10 * f + 22)
a, b, *c = [1, 2, 3, 4, 5]
*a, *b, *foo, *x = 12
*_ = 12
