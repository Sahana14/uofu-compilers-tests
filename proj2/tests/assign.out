(Module
 (Assign (targets (Name a)) (value (Name b)))
 (Assign (targets (Tuple (Name a) (Name b))) (value (Name c)))
 (Assign (targets (Tuple (Name a))) (value (Name c)))
 (Assign (targets (Tuple (Name a) (Name b))) (value (Name c)))
 (Assign (targets (Name a)) (value (Yield)))
 (Assign (targets (Name a)) (value (YieldFrom (Name baz))))
 (Assign (targets (Tuple (Name a) (Name b) (Name c))) (value (Yield)))
 (Assign
  (targets (Tuple (Name a) (Name b) (Name c)))
  (value (YieldFrom (Name baz))))
 (Assign
  (targets (Name a))
  (value
   (Call
    (func (Name foo))
    (args (BinOp (BinOp (Num 10) Mult (Name f)) Add (Num 22)))
    (keywords)
    (starargs #f)
    (kwargs #f))))
 (Assign
  (targets (Tuple (Name a) (Name b) (Starred (Name c))))
  (value (List (Num 1) (Num 2) (Num 3) (Num 4) (Num 5))))
 (Assign
  (targets
   (Tuple
    (Starred (Name a))
    (Starred (Name b))
    (Starred (Name foo))
    (Starred (Name x))))
  (value (Num 12)))
 (Assign (targets (Starred (Name _))) (value (Num 12))))
